### PR TITLE
fix book review tag and update test cases https://github.com/ubtue/Da…

### PR DESCRIPTION
**Pull request by socheres**
…tenProbleme/issues/1533

### CODE REVIEW

File: Wiley Online Library.js [hunk 1]
```diff
@@ -85,8 +85,8 @@ function processSubtitles(doc, item) {
 function addBookReviewTag(doc, item) {
 	var primaryHeading = ZU.xpathText(doc, '//span[@class="primary-heading"]');
 	if (primaryHeading) {
-		primaryHeading = primaryHeading.trim().replace(/book\s?review.*/i, 'Book Review');
-		if (primaryHeading.match(/(Book Review)|(Review Essays?)|(reviews?)/))
+		primaryHeading = primaryHeading.trim().replace(/(book\s?review.*)|(reviews?.*)|(review\sessays?.*)/i, 'Book Review');
```
Comment (jriedl) [inline Wiley Online Library.js:88]: Looks fairly complicated: Why isn't 
 ```js
if (primaryHeading.match(/Book Review|Review Essays?|Reviews?/i))
      item.tags.push('Book Review');
```
sufficient?

**Review by socheres — COMMENTED** [review]
https://github.com/ubtue/DatenProbleme/issues/1535

**Comment by socheres** [comment]
https://github.com/ubtue/DatenProbleme/issues/1535

File: Wiley Online Library.js (hunk 1 above)
Comment (socheres) [reply Wiley Online Library.js:71]: No. If the output is e.g. "review, review", i want to clean unwanted double tag

**Review by socheres — COMMENTED** [review]
here the commit link https://github.com/socheres/translators/commit/87e78444a922ede6e81ff74004967b8e7bd920a8/#diff-c4abffeb6d77d14d54a47ae5755f45de131274e3e7215ea81a7d9d3c71abc8ea

File: Wiley Online Library.js (hunk 1 above)
Comment (jriedl) [reply Wiley Online Library.js:71]: But in effect you replace the whole expression by "Book Review" then test the match again in  line 89f and add to tags afterwards. The original value of `primary heading` is not kept.

Comment (socheres) [reply Wiley Online Library.js:71]: Oh, yes. Line 88 should be after line 89.

Comment (jriedl) [reply Wiley Online Library.js]: But still with the current version, all "review-related" things are replaced by 'Book Review', all other results are inserted to tags unconditionally, although they are possibly entirely unrelated and the function is called `addBookReviewTag()`

Comment (socheres) [reply Wiley Online Library.js]: review-related tags should be intentionally replaced by "Book Review" to create "Rezension" as GND.

Comment (jriedl) [reply Wiley Online Library.js]: Yes, of course. So why not just stick to that in this function, i.e. generate a "Book Review" tag if appropriate and handle possible other keywords - if they can arise at all - at some other place.

Comment (socheres) [reply Wiley Online Library.js]: primaryHeading is just a category, not keywords. so you don't have to check additionally if there are other keywords.

**Comment by jriedl** [comment]
In Kürze auf nu.

**Pull request merged**